### PR TITLE
Rename APM tracking feature flags to tracing and enable features by default

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Rename APM tracking feature flags to tracing and enable features by default

## :bulb: Motivation and Context
Closes https://github.com/getsentry/sentry-dart/issues/1177

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Review by brustolin — APPROVED** [review]
LGTM

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 1]
```diff
@@ -168,8 +167,9 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
         val name = sdk["name"] as? String
         val version = sdk["version"] as? String
         if (name != null && version != null) {
+          val sdkVersion = SdkVersion(name, version)
           options.setSentryClientName(name)
-          options.setSdkVersion(version)
+          options.setSdkVersion(sdkVersion)
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: @denrase check this.

Comment (denrase) [reply flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: I'm getting an issue when trying to import this

Bildschirm­foto 2023-01-16 um 10 45 49

Using `PrivateSentrySDKOnly.setSdkName(name, andVersionString: version)` instead.

Comment (denrase) [reply flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: Nevermind, this is kotlin, we have the other API on iOS.

File: flutter/ios/sentry_flutter.podspec [hunk 2]
```diff
@@ -13,11 +13,11 @@ Sentry SDK for Flutter with support to native through sentry-cocoa.
   s.source_files     = 'Classes/**/*'
   s.public_header_files = 'Classes/**/*.h'
   s.dependency 'Sentry/HybridSDK', '8.0.0-rc.1'
```
Comment (denrase) [inline flutter/ios/sentry_flutter.podspec]: We should remove the duplicated dependency line here, even though its the same now.

```
s.dependency 'Sentry/HybridSDK', '8.0.0-rc.1'
s.dependency 'Sentry/HybridSDK', '8.0.0-rc.1'
```

**Review by denrase — APPROVED** [review]
One comment, other than that LGTM. Also updated the other PR now to build.

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀

**Pull request merged**